core/src/providers/mistral.rs

```diff
@@ -150,15 +151,19 @@ impl TryFrom<&ChatMessage> for MistralChatMessage {
         match cm {
             ChatMessage::Assistant(assistant_msg) => Ok(MistralChatMessage {
                 role: MistralChatMessageRole::Assistant,
-                content: assistant_msg.content.clone(),
-                tool_calls: match assistant_msg.function_calls.as_ref() {
-                    Some(fc) => Some(
+                content: match assistant_msg.clone().function_calls {
```

Comment:
are we sure we need this clone ? seems a bit weird

Reply:
Got rid of it 👌🏼